### DISCUSSION TITLE
Install MXNET 1.5.X on Raspberry Pi 4

### DISCUSSION THREAD

**Discussion by rsinventor** [apache/incubator-mxnet#20151] [Ideas]
We must document the MXNET installation steps properly in Readme file. 

The recommended way is to do this on any Linux machine with 16 GB memory and 8 CPU machine using Docker. It takes 15 mins to finish the build. Native building on raspberry itself is possible but takes more than 4-5 hours.

To install follow this link which has instructions but are not clear

  [https://mxnet.apache.org/versions/1.4.1/install/index.html?version=master&platform=Devices&language=Python&processor=CPU](https://mxnet.apache.org/versions/1.4.1/install/index.html?version=master&platform=Devices&language=Python&processor=CPU
      )

**Docker installation**
Step 1 Install Docker on your machine by following the docker installation instructions.

Note - You can install [Community Edition (CE)
](https://docs.docker.com/engine/installation/linux/ubuntu/#install-using-the-repository)
Step 2 Post installation steps to manage [Docker] as a non-root user.

Follow the four steps in this docker [documentation](https://docs.docker.com/engine/installation/linux/linux-postinstall/#manage-docker-as-a-non-root-user) to allow managing docker containers without sudo.

   
**Actual Build**

Step 1: Go to folder of your choice e.g.

`cd ~/Downloads` 

Step 2: Clone the repository e.g. targeting MXNET verison 1.5.x, using this command

`git clone -b v1.5.x https://github.com/apache/incubator-mxnet.git mxnet`

Step 3: go inside mxnet that you cloned

`cd mxnet`

Step 4: run this command to also get [additional 3rd party ](https://github.com/apache/incubator-mxnet/issues/15935)repos for successful build

`git submodule update --init --recursive'

Step5: Run the build using docker by kicking of the python script

`ci/build.py -p armv7`

 The resulting artifact will be located in build/mxnet-x.x.x-py2.py3-none-any.whl, copy this file to your Raspberry Pi. For us the target file name was **mxnet-1.5.1-py2.py3-none-any.whl**

You will also need to do this command on RaspberryPi

`pip install mxnet-1.5.1-py2.py3-none-any.whl`

`sudo apt-get install libgfortran3`

`pip install --upgrade requests==2.20.1`

`pip install gluoncv==0.7.0`

`pip install opencv-python`

**Comment by szha** [comment]
We have [a guide on building for raspberry pi](https://mxnet.apache.org/versions/master/get_started?platform=devices&iot=raspberry-pi&) ([source](https://github.com/apache/incubator-mxnet/blob/master/docs/static_site/src/_includes/get_started/devices/raspberry_pi.md)). This doc includes a guide on cross-compilation. Have you tried the instructions there? Do you see a need to update the doc?

**Reply by rsinventor** [reply]
@szha Yes! The instructions are very outdated and not beginner-friendly. I recommend updating it as per the above instructions. Also, it would make life easier if the compiled binary for various MxNet versions for raspberry pi and other 32-bit systems was provided at the PyPi project.
